packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/AppImpl.kt

```diff
@@ -127,4 +129,13 @@ public class AppImpl(
         networkTransport.close()
         nativePointer.release()
     }
+
+    internal companion object {
+        // This method is used to inject bundleId to the sync configuration. The
+        // SyncLoweringExtension is replacing calls to App.create(appId) with calls to this method.
+        internal fun create(appId: String, bundleId: String): App {
+            Validation.checkEmpty(appId, "appId")
+            return App.create(AppConfiguration.Builder(appId).build(bundleId))
+        }
+    }
```

Comment:
But `App` and `AppImpl` is in our library, so we cannot inject anything to that. If we just reference a `const` here then we cannot have multiple modules, because we would get overlapping symbols. I guess we could try to put this into its own separate builder method, so we don't rewire a user call, but I can't really see how we can do it when we have the various `create` shortcuts for `App.create(AppConfiguraiton.Builder(appId).build(...))`.

Reply:
Ah yeah, good point 🙈 . Ignore the comment.

Reply:
I just tried to trigger the exception by a call to 
```
419:     val app = App.create("")
```
And even though we rewire the call the stacktrace is still pointing correctly to the call site of the original call, so I would deem it acceptable ... ?
```
	at io.realm.kotlin.mongodb.internal.AppImpl$Companion.create$io_realm_kotlin_library(AppImpl.kt:137)
	at io.realm.kotlin.test.mongodb.shared.AppConfigurationTests.injectedBundleId(AppConfigurationTests.kt:419)
```

Reply:
Yeah, I guess that is fine 👍